"""Multi-Language AI Coordinator
å¤šèªè¨€ AI å”èª¿å™¨ - V2 gRPC çµ±ä¸€æ¶æ§‹

è² è²¬å”èª¿ Python/Rust/Go/TypeScript ç­‰å¤šèªè¨€ AI æ¨¡çµ„
ä½¿ç”¨çµ±ä¸€çš„ gRPC æ¡†æ¶é€²è¡Œè·¨èªè¨€é€šè¨Šï¼Œæ›¿ä»£èˆŠç‰ˆ HTTP/subprocess æ–¹å¼

æ•´åˆåŠŸèƒ½ï¼š
- Rust æ¨¡çµ„ï¼šé«˜æ•ˆèƒ½æƒæå’Œè¨ˆç®—
- Go æ¨¡çµ„ï¼šä¸¦ç™¼è™•ç†å’Œå¾®æœå‹™
- TypeScript æ¨¡çµ„ï¼šå‹•æ…‹åˆ†æå’Œå‰ç«¯æ•´åˆ
- Python æ ¸å¿ƒï¼šAI æ±ºç­–å’Œç­–ç•¥è¦åŠƒ
"""

import asyncio
import time
from typing import Any

from services.aiva_common.cross_language.core import CrossLanguageService, CrossLanguageConfig
from services.aiva_common.cross_language.adapters import (
    RustAdapter, RustConfig, create_rust_adapter,
    GoAdapter, GoConfig, create_go_adapter
)
from services.aiva_common.enums.modules import ProgrammingLanguage

from .utils.logging_formatter import get_aiva_logger, log_cross_language_call

logger = get_aiva_logger("multilang_coordinator")


class MultiLanguageAICoordinator:
    """å¤šèªè¨€ AI å”èª¿å™¨ - V2 gRPC çµ±ä¸€ç‰ˆæœ¬"""

    def __init__(self, config: CrossLanguageConfig | None = None):
        self.available_ai_modules: dict[ProgrammingLanguage, bool] = {
            ProgrammingLanguage.PYTHON: True,  # ä¸»è¦ AI å¼•æ“
            ProgrammingLanguage.RUST: False,  # Rust AI æ¨¡çµ„ï¼ˆgRPC æœå‹™ï¼‰
            ProgrammingLanguage.GO: False,  # Go AI æ¨¡çµ„ï¼ˆgRPC æœå‹™ï¼‰
            ProgrammingLanguage.TYPESCRIPT: False,  # TypeScript AI æ¨¡çµ„ï¼ˆgRPC æœå‹™ï¼‰
        }
        self.module_status: dict[str, Any] = {}
        
        # V2 çµ±ä¸€ gRPC æœå‹™
        self.cross_lang_service = CrossLanguageService(config)
        
        # èªè¨€é©é…å™¨å¯¦ä¾‹
        self.rust_adapter: RustAdapter | None = None
        self.go_adapter: GoAdapter | None = None
        self.typescript_adapter: Any | None = None  # TypeScript adapter placeholder
        
        # gRPC æœå‹™ç«¯é»é…ç½®
        self.service_endpoints = {
            ProgrammingLanguage.RUST: "localhost:50052",
            ProgrammingLanguage.GO: "localhost:50053", 
            ProgrammingLanguage.TYPESCRIPT: "localhost:50054"
        }

        # éåŒæ­¥åˆå§‹åŒ–ï¼ˆéœ€è¦åœ¨ç•°æ­¥ä¸Šä¸‹æ–‡ä¸­èª¿ç”¨ï¼‰
        self._initialized = False

    async def initialize(self):
        """ç•°æ­¥åˆå§‹åŒ– gRPC é€£æ¥å’Œæ¨¡çµ„æª¢æŸ¥"""
        if self._initialized:
            return
            
        logger.info("æ­£åœ¨åˆå§‹åŒ–å¤šèªè¨€ AI å”èª¿å™¨...")
        
        # å‰µå»º Rust é©é…å™¨
        try:
            self.rust_adapter = await create_rust_adapter(RustConfig())
            if await self.rust_adapter.initialize():
                self.available_ai_modules[ProgrammingLanguage.RUST] = True
                logger.info("âœ… Rust æ¨¡çµ„åˆå§‹åŒ–æˆåŠŸ")
        except Exception as e:
            logger.warning(f"âš ï¸ Rust æ¨¡çµ„åˆå§‹åŒ–å¤±æ•—: {e}")
        
        # å‰µå»º Go é©é…å™¨
        try:
            self.go_adapter = await create_go_adapter(GoConfig())
            if await self.go_adapter.initialize():
                self.available_ai_modules[ProgrammingLanguage.GO] = True
                logger.info("âœ… Go æ¨¡çµ„åˆå§‹åŒ–æˆåŠŸ")
        except Exception as e:
            logger.warning(f"âš ï¸ Go æ¨¡çµ„åˆå§‹åŒ–å¤±æ•—: {e}")
        
        # æª¢æŸ¥ TypeScript æœå‹™
        await self._check_typescript_service()
        
        self._initialized = True
        
        # è¼¸å‡ºåˆå§‹åŒ–ç‹€æ…‹
        available_langs = [lang.value for lang, available in self.available_ai_modules.items() if available]
        logger.info(f"âœ… å¤šèªè¨€ AI å”èª¿å™¨åˆå§‹åŒ–å®Œæˆ - å¯ç”¨èªè¨€: {', '.join(available_langs)}")

    def check_module_availability(self, language: ProgrammingLanguage) -> bool:
        """æª¢æŸ¥ç‰¹å®šèªè¨€çš„ AI æ¨¡çµ„æ˜¯å¦å¯ç”¨"""
        return self.available_ai_modules.get(language, False)

    async def execute_task(
        self, task: str, language: ProgrammingLanguage | None = None, **kwargs
    ) -> dict[str, Any]:
        """åŸ·è¡Œ AI ä»»å‹™

        Args:
            task: ä»»å‹™é¡å‹
            language: æŒ‡å®šä½¿ç”¨çš„èªè¨€ï¼ˆNone å‰‡è‡ªå‹•é¸æ“‡ï¼‰
            **kwargs: ä»»å‹™åƒæ•¸

        Returns:
            ä»»å‹™åŸ·è¡Œçµæœ
        """
        if language is None:
            # è‡ªå‹•é¸æ“‡å¯ç”¨çš„èªè¨€æ¨¡çµ„
            language = self._select_best_language(task)

        logger.info(f"åŸ·è¡Œ AI ä»»å‹™: {task}, ä½¿ç”¨èªè¨€: {language}")

        # æ ¹æ“šèªè¨€èª¿ç”¨å°æ‡‰çš„æ¨¡çµ„
        try:
            if language == ProgrammingLanguage.RUST:
                return await self.call_rust_ai(task, **kwargs)
            elif language == ProgrammingLanguage.GO:
                return await self.call_go_ai(task, **kwargs)
            elif language == ProgrammingLanguage.TYPESCRIPT:
                return await self.call_typescript_ai(task, **kwargs)
            elif language == ProgrammingLanguage.PYTHON:
                # Python æœ¬åœ°è™•ç†
                return await self._execute_python_task(task, **kwargs)
            else:
                return {
                    "success": False,
                    "error": f"ä¸æ”¯æŒçš„èªè¨€: {language}",
                    "language": language,
                }
        except Exception as e:
            logger.error(f"ä»»å‹™åŸ·è¡Œç•°å¸¸: {e}")
            return {
                "success": False,
                "error": str(e),
                "task": task,
                "language": language,
            }

    async def _execute_python_task(self, task: str, **kwargs) -> dict[str, Any]:
        """åŸ·è¡Œ Python AI ä»»å‹™"""
        logger.info(f"åŸ·è¡Œ Python AI ä»»å‹™: {task}")

        # é€™è£¡å¯ä»¥èª¿ç”¨æœ¬åœ°çš„ Python AI åŠŸèƒ½
        # ä¾‹å¦‚ BioNeuronRAGAgent æˆ–å…¶ä»– AI çµ„ä»¶

        # æ¨¡æ“¬è™•ç†
        import asyncio

        await asyncio.sleep(0.1)  # æ¨¡æ“¬è™•ç†æ™‚é–“

        return {
            "success": True,
            "task": task,
            "language": "python",
            "result": f"Python AI ä»»å‹™ '{task}' åŸ·è¡Œå®Œæˆ",
            "details": kwargs,
            "processed_by": "MultiLanguageAICoordinator",
        }

    def _select_best_language(self, task: str) -> ProgrammingLanguage:
        """æ ¹æ“šä»»å‹™é¸æ“‡æœ€ä½³èªè¨€"""
        # å„ªå…ˆä½¿ç”¨ Pythonï¼ˆä¸»è¦ AI å¼•æ“ï¼‰
        if self.available_ai_modules[ProgrammingLanguage.PYTHON]:
            return ProgrammingLanguage.PYTHON

        # æ€§èƒ½å¯†é›†å‹ä»»å‹™å„ªå…ˆä½¿ç”¨ Rust
        performance_intensive = ["vulnerability_scan", "fuzzing", "exploit"]
        if any(keyword in task.lower() for keyword in performance_intensive):
            if self.available_ai_modules[ProgrammingLanguage.RUST]:
                return ProgrammingLanguage.RUST

        # ä½µç™¼ä»»å‹™å„ªå…ˆä½¿ç”¨ Go
        concurrent_tasks = ["parallel", "distributed", "concurrent"]
        if any(keyword in task.lower() for keyword in concurrent_tasks):
            if self.available_ai_modules[ProgrammingLanguage.GO]:
                return ProgrammingLanguage.GO

        # é»˜èªä½¿ç”¨ Python
        return ProgrammingLanguage.PYTHON

    def get_status(self) -> dict[str, Any]:
        """ç²å–å”èª¿å™¨ç‹€æ…‹"""
        return {
            "available_modules": {
                lang.value: available
                for lang, available in self.available_ai_modules.items()
            },
            "module_status": self.module_status,
        }

    def enable_module(self, language: ProgrammingLanguage) -> bool:
        """å•Ÿç”¨ç‰¹å®šèªè¨€æ¨¡çµ„"""
        try:
            self.available_ai_modules[language] = True
            logger.info(f"å·²å•Ÿç”¨ {language} AI æ¨¡çµ„")
            return True
        except Exception as e:
            logger.error(f"å•Ÿç”¨ {language} æ¨¡çµ„å¤±æ•—: {e}")
            return False

    def disable_module(self, language: ProgrammingLanguage) -> bool:
        """ç¦ç”¨ç‰¹å®šèªè¨€æ¨¡çµ„"""
        try:
            self.available_ai_modules[language] = False
            logger.info(f"å·²ç¦ç”¨ {language} AI æ¨¡çµ„")
            return True
        except Exception as e:
            logger.error(f"ç¦ç”¨ {language} æ¨¡çµ„å¤±æ•—: {e}")
            return False

    async def _check_rust_service(self) -> None:
        """æª¢æŸ¥ Rust gRPC æœå‹™å¯ç”¨æ€§"""
        try:
            logger.info("æ­£åœ¨æª¢æŸ¥ Rust gRPC æœå‹™...")
            endpoint = self.service_endpoints[ProgrammingLanguage.RUST]
            
            # ä½¿ç”¨ V2 gRPC å¥åº·æª¢æŸ¥
            is_available = await self.cross_lang_service.health_check(endpoint)
            
            if is_available:
                self.available_ai_modules[ProgrammingLanguage.RUST] = True
                logger.info("âœ… Rust gRPC æœå‹™å·²å°±ç·’")
                self.module_status[ProgrammingLanguage.RUST] = {
                    "status": "ready",
                    "endpoint": endpoint,
                    "protocol": "gRPC",
                    "checked_at": time.time(),
                }
            else:
                logger.info("ğŸ“ Rust gRPC æœå‹™æœªé‹è¡Œï¼Œä¿æŒç¦ç”¨ç‹€æ…‹")
                self.module_status[ProgrammingLanguage.RUST] = {
                    "status": "unavailable", 
                    "endpoint": endpoint,
                    "protocol": "gRPC"
                }

        except Exception as e:
            logger.error(f"Rust æœå‹™æª¢æŸ¥ç•°å¸¸: {e}")
            self.available_ai_modules[ProgrammingLanguage.RUST] = False

    async def _check_go_service(self) -> None:
        """æª¢æŸ¥ Go gRPC æœå‹™å¯ç”¨æ€§"""
        try:
            logger.info("æ­£åœ¨æª¢æŸ¥ Go gRPC æœå‹™...")
            endpoint = self.service_endpoints[ProgrammingLanguage.GO]
            
            # ä½¿ç”¨ V2 gRPC å¥åº·æª¢æŸ¥
            is_available = await self.cross_lang_service.health_check(endpoint)
            
            if is_available:
                self.available_ai_modules[ProgrammingLanguage.GO] = True
                logger.info("âœ… Go gRPC æœå‹™å·²å°±ç·’")
                self.module_status[ProgrammingLanguage.GO] = {
                    "status": "ready",
                    "endpoint": endpoint,
                    "protocol": "gRPC",
                    "checked_at": time.time(),
                }
            else:
                logger.info("ğŸ“ Go gRPC æœå‹™æœªé‹è¡Œï¼Œä¿æŒç¦ç”¨ç‹€æ…‹")
                self.module_status[ProgrammingLanguage.GO] = {
                    "status": "unavailable",
                    "endpoint": endpoint,
                    "protocol": "gRPC"
                }

        except Exception as e:
            logger.error(f"Go æœå‹™æª¢æŸ¥ç•°å¸¸: {e}")
            self.available_ai_modules[ProgrammingLanguage.GO] = False

    async def _check_typescript_service(self) -> None:
        """æª¢æŸ¥ TypeScript gRPC æœå‹™å¯ç”¨æ€§"""
        try:
            logger.info("æ­£åœ¨æª¢æŸ¥ TypeScript gRPC æœå‹™...")
            endpoint = self.service_endpoints[ProgrammingLanguage.TYPESCRIPT]
            
            # ä½¿ç”¨ V2 gRPC å¥åº·æª¢æŸ¥
            is_available = await self.cross_lang_service.health_check(endpoint)
            
            if is_available:
                self.available_ai_modules[ProgrammingLanguage.TYPESCRIPT] = True
                logger.info("âœ… TypeScript gRPC æœå‹™å·²å°±ç·’")
                self.module_status[ProgrammingLanguage.TYPESCRIPT] = {
                    "status": "ready",
                    "endpoint": endpoint,
                    "protocol": "gRPC",
                    "checked_at": time.time(),
                }
            else:
                logger.info("ğŸ“ TypeScript gRPC æœå‹™æœªé‹è¡Œï¼Œä¿æŒç¦ç”¨ç‹€æ…‹")
                self.module_status[ProgrammingLanguage.TYPESCRIPT] = {
                    "status": "unavailable",
                    "endpoint": endpoint,
                    "protocol": "gRPC"
                }

        except Exception as e:
            logger.error(f"TypeScript æœå‹™æª¢æŸ¥ç•°å¸¸: {e}")
            self.available_ai_modules[ProgrammingLanguage.TYPESCRIPT] = False

    async def call_rust_ai(self, task: str, **kwargs) -> dict[str, Any]:
        """èª¿ç”¨ Rust AI æ¨¡çµ„ - ä½¿ç”¨ RustAdapter"""
        if not self.available_ai_modules[ProgrammingLanguage.RUST]:
            return {"success": False, "error": "Rust AI æ¨¡çµ„æœªå•Ÿç”¨"}

        if not self.rust_adapter:
            return {"success": False, "error": "Rusté©é…å™¨æœªåˆå§‹åŒ–"}

        start_time = time.time()
        try:
            # ç¢ºä¿å·²åˆå§‹åŒ–
            if not self._initialized:
                await self.initialize()
            
            # é€šéé©é…å™¨åŸ·è¡Œä»»å‹™ï¼ˆç•¶å‰ç‚ºä½”ä½ç¬¦å¯¦ç¾ï¼‰
            # TODO: å¯¦ç¾å®Œæ•´çš„ RustAdapter.execute_task æ–¹æ³•
            # ç›®å‰è¿”å›æ¨¡æ“¬çµæœ
            import asyncio
            await asyncio.sleep(0)  # ç¢ºä¿å‡½æ•¸æ˜¯ç•°æ­¥çš„
            
            result = {
                "task": task,
                "status": "completed",
                "data": kwargs
            }

            log_cross_language_call(
                logger,
                "python",
                "rust",
                task,
                kwargs,
                result,
                None,
                time.time() - start_time,
            )
            
            return {
                "success": True,
                "language": "rust",
                "task": task,
                "result": result,
                "protocol": "adapter"
            }

        except Exception as e:
            error_msg = f"Rusté©é…å™¨èª¿ç”¨ç•°å¸¸: {e}"
            log_cross_language_call(
                logger,
                "python",
                "rust", 
                task,
                kwargs,
                None,
                error_msg,
                time.time() - start_time,
            )
            logger.error(f"èª¿ç”¨ Rust é©é…å™¨ç•°å¸¸: {e}")
            return {"success": False, "error": error_msg}

    async def call_go_ai(self, task: str, **kwargs) -> dict[str, Any]:
        """èª¿ç”¨ Go AI æ¨¡çµ„ - V2 gRPC ç‰ˆæœ¬"""
        if not self.available_ai_modules[ProgrammingLanguage.GO]:
            return {"success": False, "error": "Go AI æ¨¡çµ„æœªå•Ÿç”¨"}

        start_time = time.time()
        try:
            # ç¢ºä¿å·²åˆå§‹åŒ–
            if not self._initialized:
                await self.initialize()
            
            # ç²å– Go gRPC æœå‹™ç«¯é»
            endpoint = self.service_endpoints[ProgrammingLanguage.GO]
            
            # æ ¹æ“šä»»å‹™é¡å‹é¸æ“‡æœå‹™
            if "data_analysis" in task.lower() or "analyze" in task.lower():
                # ä½¿ç”¨æ•¸æ“šåˆ†ææœå‹™
                from services.aiva_common.protocols.aiva_services_pb2 import DataAnalysisRequest  # type: ignore[attr-defined]
                from services.aiva_common.protocols.aiva_services_pb2_grpc import DataAnalyzerStub  # type: ignore[attr-defined]
                
                request = DataAnalysisRequest(
                    analysis_id=kwargs.get("analysis_id", f"go_{int(time.time())}"),
                    data_source=kwargs.get("data_source", ""),
                    analysis_type=task,
                    parameters=kwargs.get("parameters", {})
                )
                
                response = await self.cross_lang_service.call_service(
                    DataAnalyzerStub, "AnalyzeData", request, endpoint
                )
                
                result = {
                    "analysis_id": response.analysis_id,  # type: ignore[attr-defined]
                    "status": response.status,  # type: ignore[attr-defined]
                    "insights_count": len(response.insights),  # type: ignore[attr-defined]
                    "summary": response.summary  # type: ignore[attr-defined]
                }
                
            elif "code" in task.lower():
                # ä½¿ç”¨ä»£ç¢¼ç”Ÿæˆæœå‹™
                from services.aiva_common.protocols.aiva_services_pb2 import CodeGenerationRequest  # type: ignore[attr-defined]
                from services.aiva_common.protocols.aiva_services_pb2_grpc import CodeGeneratorStub  # type: ignore[attr-defined]
                
                request = CodeGenerationRequest(
                    generation_id=kwargs.get("generation_id", f"go_{int(time.time())}"),
                    template_type=kwargs.get("template_type", "standard"),
                    target_language=kwargs.get("target_language", "go"),
                    parameters=kwargs.get("parameters", {}),
                    specification=kwargs.get("specification", task)
                )
                
                response = await self.cross_lang_service.call_service(
                    CodeGeneratorStub, "GenerateCode", request, endpoint
                )
                
                result = {
                    "generation_id": response.generation_id,  # type: ignore[attr-defined]
                    "status": response.status,  # type: ignore[attr-defined]
                    "files_count": len(response.files),  # type: ignore[attr-defined]
                    "warnings": list(response.warnings)  # type: ignore[attr-defined]
                }
                
            else:
                # é»˜èª AI æ¨ç†
                from services.aiva_common.protocols.aiva_services_pb2 import ReasoningRequest  # type: ignore[attr-defined]
                from services.aiva_common.protocols.aiva_services_pb2_grpc import AIServiceStub  # type: ignore[attr-defined]
                
                request = ReasoningRequest(
                    query=kwargs.get("query", task),
                    session_id=kwargs.get("session_id", "default"),
                    context_items=kwargs.get("context", [])
                )
                
                response = await self.cross_lang_service.call_service(
                    AIServiceStub, "ExecuteReasoning", request, endpoint
                )
                
                result = {
                    "response": response.response,  # type: ignore[attr-defined]
                    "confidence": response.confidence,  # type: ignore[attr-defined]
                    "reasoning_steps": list(response.reasoning_steps)  # type: ignore[attr-defined]
                }

            log_cross_language_call(
                logger,
                "python",
                "go", 
                task,
                kwargs,
                result,
                None,
                time.time() - start_time,
            )
            
            return {
                "success": True,
                "language": "go",
                "task": task,
                "result": result,
                "protocol": "gRPC"
            }

        except Exception as e:
            error_msg = f"gRPC èª¿ç”¨ç•°å¸¸: {e}"
            log_cross_language_call(
                logger,
                "python",
                "go",
                task,
                kwargs,
                None,
                error_msg,
                time.time() - start_time,
            )
            logger.error(f"èª¿ç”¨ Go gRPC æœå‹™ç•°å¸¸: {e}")
            return {"success": False, "error": error_msg}

    async def call_typescript_ai(self, task: str, **kwargs) -> dict[str, Any]:
        """èª¿ç”¨ TypeScript AI æ¨¡çµ„ - V2 gRPC ç‰ˆæœ¬"""
        if not self.available_ai_modules[ProgrammingLanguage.TYPESCRIPT]:
            return {"success": False, "error": "TypeScript AI æ¨¡çµ„æœªå•Ÿç”¨"}

        start_time = time.time()
        try:
            # ç¢ºä¿å·²åˆå§‹åŒ–
            if not self._initialized:
                await self.initialize()
            
            # ç²å– TypeScript gRPC æœå‹™ç«¯é»
            endpoint = self.service_endpoints[ProgrammingLanguage.TYPESCRIPT]
            
            # æ ¹æ“šä»»å‹™é¡å‹é¸æ“‡æœå‹™ï¼ˆTypeScript ä¸»è¦ç”¨æ–¼ Web ç›¸é—œä»»å‹™ï¼‰
            if "web" in task.lower() or "http" in task.lower():
                # ä½¿ç”¨ Web æœå‹™
                from services.aiva_common.protocols.aiva_services_pb2 import ScanRequest, ScanConfig  # type: ignore[attr-defined]
                from services.aiva_common.protocols.aiva_services_pb2_grpc import WebServiceStub  # type: ignore[attr-defined]
                
                config = ScanConfig(
                    max_depth=kwargs.get("max_depth", 5),
                    timeout_seconds=kwargs.get("timeout", 30),
                    aggressive_mode=kwargs.get("aggressive", False)
                )
                
                request = ScanRequest(
                    scan_id=kwargs.get("scan_id", f"ts_{int(time.time())}"),
                    target=kwargs.get("target", kwargs.get("url", "")),
                    scan_type=task,
                    config=config
                )
                
                # ä½¿ç”¨æµå¼æƒæï¼ˆé©ç”¨æ–¼ Web æƒæï¼‰
                async for web_result in self.cross_lang_service.call_service(  # type: ignore[misc]
                    WebServiceStub, "ScanWebsite", request, endpoint
                ):
                    # è™•ç†æµå¼çµæœï¼ˆé€™è£¡ç°¡åŒ–ç‚ºå–ç¬¬ä¸€å€‹çµæœï¼‰
                    result = {
                        "scan_id": web_result.scan_id,  # type: ignore[attr-defined]
                        "request_info": {
                            "method": web_result.request.method,  # type: ignore[attr-defined]
                            "url": web_result.request.url  # type: ignore[attr-defined]
                        },
                        "response_info": {
                            "status_code": web_result.response.status_code,  # type: ignore[attr-defined]
                            "response_time": web_result.response.response_time_ms  # type: ignore[attr-defined]
                        },
                        "findings_count": len(web_result.findings)  # type: ignore[attr-defined]
                    }
                    break  # å–ç¬¬ä¸€å€‹çµæœä½œç‚ºç¤ºä¾‹
                    
            else:
                # é»˜èªä½¿ç”¨ AI æœå‹™é€²è¡Œå‘½ä»¤åˆ†æ
                from services.aiva_common.protocols.aiva_services_pb2 import CommandAnalysisRequest  # type: ignore[attr-defined]
                from services.aiva_common.protocols.aiva_services_pb2_grpc import AIServiceStub  # type: ignore[attr-defined]
                
                request = CommandAnalysisRequest(
                    command=task,
                    session_id=kwargs.get("session_id", "ts_session"),
                    context=kwargs.get("context", {}),
                    user_id=kwargs.get("user_id", "default")
                )
                
                response = await self.cross_lang_service.call_service(
                    AIServiceStub, "AnalyzeCommand", request, endpoint
                )
                
                result = {
                    "intent": response.intent,  # type: ignore[attr-defined]
                    "recommended_service": response.recommended_service,  # type: ignore[attr-defined]
                    "parameters": dict(response.parameters),  # type: ignore[attr-defined]
                    "confidence": response.confidence,  # type: ignore[attr-defined]
                    "suggestions": list(response.suggestions)  # type: ignore[attr-defined]
                }

            log_cross_language_call(
                logger,
                "python",
                "typescript",
                task,
                kwargs,
                result,
                None,
                time.time() - start_time,
            )
            
            return {
                "success": True,
                "language": "typescript",
                "task": task,
                "result": result,
                "protocol": "gRPC"
            }

        except Exception as e:
            error_msg = f"gRPC èª¿ç”¨ç•°å¸¸: {e}"
            log_cross_language_call(
                logger,
                "python",
                "typescript",
                task,
                kwargs,
                None,
                error_msg,
                time.time() - start_time,
            )
            logger.error(f"èª¿ç”¨ TypeScript gRPC æœå‹™ç•°å¸¸: {e}")
            return {"success": False, "error": error_msg}
